Applications/Spire/Include/Spire/Ui/DecimalInputWidget.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef SPIRE_DECIMAL_INPUT_WIDGET_HPP
+#define SPIRE_DECIMAL_INPUT_WIDGET_HPP
+#include "Spire/Ui/NumericInputWidget.hpp"
+
+namespace Spire {
+
+  //! Represents a widget for displaying and modifying decimal numbers.
+  class DecimalInputWidget : public QWidget {
```

Comment:
As specified, this should inherit from ```QSpinBox```, ```QDoubleSpinBox```, or ```QAbstractSpinBox```.

Reply:
We discussed inheriting the QAbstractSpinBox for the NumericInputWidget, then using composition to create each type of numeric widget. If I'm subclassing for each type then I have to re-implement the editing and validation for each type.

Reply:
Can you reference where this was discussed?

Reply:
The 'Implement Quantity Input Widget' comments on August 17th.

Reply:
I based these implementations on my suggestion:

"If I'm going to subclass QAbstractSpinBox for the QuantityInputWidget, it might
make sense to create a NumericInputWidget and have each spin box type
subclass that implementation. I worry about having to create 3 different
implementations that all reimplement mostly the same UX flow, when it's
only the acceptable input formats that differ from widget-to-widget."

and your reply:

"Sure you can do that, have the NumericInputWidget inherit from
QAbstractSpinBox and operate on strings or it might be better to factor out
the common functionality into NumericInputWidget and use composition
instead."

I'm not sure how else to interpret that. If I have to go back and subclass the other spin box types (QSpinBox, QDoubleSpinBox) then I'm effectively re-implementing the editing and validating behavior multiple times.